packages/codemirror-blocks/src/ui/PrimitiveList.tsx

```diff
@@ -18,65 +18,64 @@ require("./PrimitiveList.less");
 type BasePrimitiveProps = {
   primitive: LanguagePrimitive;
   className: string;
-  onFocus: (primitive: LanguagePrimitive) => void;
-  onKeyDown: (e: React.KeyboardEvent) => void;
+  onFocus: (e: React.FocusEvent) => void;
 };
 
-export const Primitive = (props: BasePrimitiveProps) => {
-  const { primitive, className, onFocus, onKeyDown } = props;
+// @pcardune - I still needed to cast the ref here, in order to satisfy TS
+export const Primitive = React.forwardRef<HTMLElement, BasePrimitiveProps>(
+  (props, ref: React.RefObject<HTMLElement>) => {
```

Comment:
You shouldn't need to. The following should be enough.

```ts
export const Primitive = React.forwardRef<HTMLSpanElement, BasePrimitiveProps>(
  (props, ref) => {
```

Reply:
Here's what I see when I do that: 
![image](https://user-images.githubusercontent.com/77126/141370545-ccf2cc88-b55f-4f17-9929-da70dce18cf3.png)